actors/builtin/verifreg/verified_registry_actor.go

```diff
@@ -318,3 +319,120 @@ func (a Actor) RestoreBytes(rt runtime.Runtime, params *RestoreBytesParams) *abi
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to get verified client %v", params.VerifiedClientToRemove)
+		if !isVerifiedClient {
+			rt.Abortf(exitcode.ErrNotFound, "%v is not a verified client", params.VerifiedClientToRemove)
+		}
+
+		// validate requesters are verifiers
+		if isVerifier, code, err := isVerifier(rt, st, verifier1); !isVerifier {
+			rt.Abortf(code, err.Error())
+		}
+		if isVerifier, code, err := isVerifier(rt, st, verifier2); !isVerifier {
+			rt.Abortf(code, err.Error())
+		}
+
+		// get the remove datacap proposal IDs for  the verifiers
+		proposalIDs, err := adt.AsMap(adt.AsStore(rt), st.RemoveDataCapProposalIDs, builtin.DefaultHamtBitwidth)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load datacap removal proposal ids")
+
+		var verifier1ID RmDcProposalID
+		idExists, err := proposalIDs.Get(abi.NewAddrPairKey(verifier1, client), &verifier1ID)
+		if !idExists { //initialize with 0
+			verifier1ID = RmDcProposalID{proposalID: 0}
+			err = proposalIDs.Put(abi.NewAddrPairKey(verifier1, client), &verifier1ID)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add remove datacap proposal id for the verifier")
+		}
+
+
+		var verifier2ID RmDcProposalID
+		idExists, err = proposalIDs.Get(abi.NewAddrPairKey(verifier2, client), &verifier2ID)
+		if !idExists { //initialize with 0
+			verifier2ID = RmDcProposalID{proposalID: 0}
+			err = proposalIDs.Put(abi.NewAddrPairKey(verifier2, client), &verifier2ID)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add remove datacap proposal id for the verifier")
+		}
+
+		// verify verifier request signature
+		removalProposal := RemoveDataCapProposal{
+			VerifiedClient: client,
+			DataCapAmount:  params.DataCapAmountToRemove,
+		}
+		if verifierSignVerified, code, err := removeDataCapRequestIsValid(rt, params.VerifierRequest1, removalProposal, verifier1ID); !verifierSignVerified {
+			rt.Abortf(code, err.Error())
+		}
+		if verifierSignVerified, code, err := removeDataCapRequestIsValid(rt, params.VerifierRequest2, removalProposal, verifier2ID); !verifierSignVerified {
+			rt.Abortf(code, err.Error())
+		}
+
+		// execute the datacap removal
+		newDataCap := big.Sub(preDataCap, params.DataCapAmountToRemove)
+		if newDataCap.LessThanEqual(big.NewInt(0)) { // no DataCap remaining
+			// delete verified client
+			err = verifiedClients.Delete(abi.AddrKey(client))
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to delete verified client %v", params.VerifiedClientToRemove)
+		} else {
+			// update the DataCap amount after the removal
+			err = verifiedClients.Put(abi.AddrKey(client), &newDataCap)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to update verified client %v to %v DataCap %v", params.VerifiedClientToRemove, newDataCap)
+			if preDataCap.LessThan(params.DataCapAmountToRemove) {
```

Comment:
I think its the other case where you want to reset this as preDataCap - DataCapAmountToRemove < 0 is the same as preDataCap < DataCapAmountToRemove